pkg/runner/client_pool.go

```diff
@@ -0,0 +1,56 @@
+package runner
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type pool struct {
+	sync.Mutex
+
+	availableC chan *kubernetes.Clientset
+}
+
+// newPool returns a pool of Kubernetes clientset connections
+func newPool(workers int, config KubernetesConfig) (*pool, error) {
+	k8scfg, err := clientcmd.BuildConfigFromFlags("", config.KubeConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not start k8s client from config: %v", err)
+	}
+
+	pool := &pool{
+		availableC: make(chan *kubernetes.Clientset, workers),
+	}
+
+	for i := 0; i < workers; i++ {
+		k8sClientset, err := kubernetes.NewForConfig(k8scfg)
+		if err != nil {
+			return nil, fmt.Errorf("could not create k8s clientset: %v", err)
+		}
+
+		pool.availableC <- k8sClientset
+	}
+
+	return pool, nil
+}
+
+func (p *pool) Get(ctx context.Context) (*kubernetes.Clientset, error) {
```

Comment:
better names for these might be `Acquire` and `Release`.

Reply:
I'm a bit confused about how 'Acquire' and 'Release' might be equivalent. If I understood correctly, this a connection pool and this get is used to acquire a connection from the pool. If so, I'd rename it acquire as it is a bit more explicit.

Reply:
OK, will rename.

The idea is that we have `n` connections, and just pick one that is `free` whenever we need a client, as this is a slow HTTP client and blocks for quite a while when we make a request to the Kubernetes API.